ignite/distributed/utils.py

```diff
@@ -321,7 +321,7 @@ def train_fn(local_rank, a, b, c, d=12):
         )
 
 
-def all_reduce(tensor: Union[torch.Tensor, float], op: str = "SUM") -> Union[torch.Tensor, float]:
+def all_reduce(tensor: Union[torch.Tensor, float], op: str = "SUM", group: Any = None) -> Union[torch.Tensor, float]:
```

Comment:
```suggestion
def all_reduce(tensor: Union[torch.Tensor, float], op: str = "SUM", group: Optional[Union[Any, List[int]]] = None) -> Union[torch.Tensor, float]:
```

Reply:
1) `all_reduce(tensor, group="abc")` -> will raise an error in _model.new_group(group) => OK

2) Below use-case should not fail:
```python
assert idist.backend() == "nccl"
group = dist.new_group([0, 1])
idist.all_reduce(tensor, group=group)
```

3) Below use-case should fail in ignite/distributed/comp_models/native.py in `_do_all_reduce`
```python
assert idist.backend() == "horovod"
group = dist.new_group([0, 1])
idist.all_reduce(tensor, group=group)
```